app-modules/engagement/src/Jobs/CreateBatchedEngagement.php

```diff
@@ -56,11 +58,26 @@ class CreateBatchedEngagement implements ShouldQueue
     use Queueable;
     use SerializesModels;
 
+    public int $maxExceptions = 3;
+
     public function __construct(
         public EngagementBatch $engagementBatch,
         public CanBeNotified $recipient,
     ) {}
 
+    /**
+     * @return array<object>
+     */
+    public function middleware(): array
+    {
+        return [new RateLimitedWithRedis('notification')];
+    }
+
+    public function retryUntil(): DateTime
```

Comment:
Please swap this out for `DateTimeInterface` so that if the `now()` implementation gets switched to being immutable, this code doesn't break